testdata/bundles/plain-v0/dependent/build-push-e2e-bundle.sh

```diff
@@ -0,0 +1,75 @@
+#! /bin/bash
```

Comment:
Instead of having these separate files sharing a lot of common scripting, can we instead just have one script with arguments and have each sub-folder contain the needed yaml?

Reply:
I originally had it as a single script that did this, but since some of the testdata has specific nuances for the tests there was an increase in the single script complexity that I felt would only continue to get more complex when/if new testdata is added. Some examples:
- Two different bundle formats, `registry+v1` and `plain+v0`, have different directory requirements. This means that a common script must account for this difference and as such introduces more complex logic to ensure there are no flakes when building each bundle.
- One of the `plain+v0` testdata bundles requires inclusion of subdirectories. this requires more complex configmap creation and volume mounting logic to ensure the subdirectory is copied over to the container running the build + push operations.
- Ability to push to a registry that requires authentication requires additional volume mounts and configuration of the container to successfully authenticate with the registry.

I felt that reducing the script complexity and increasing "repetition" is likely going to be more maintainable in the long run since additional test cases can get as custom as they need to and provide all the logic to ensure the image is appropriately built and pushed to the on cluster registry.

Reply:
Is there a way to reduce down to a single script and make the specific nuances encapsulated with some other mechanism? For example, it seems like the nuances are in the shape of the bundles. Can we tar.gz the root directories and then have the script deal in tar.gz files generically. Configmap data is tar.gz, and the build/push pod is changed to untar prior to running the build?

Reply:
I'm sure we could do _something_ to make it one or two scripts (likely 2 for the pushing to a registry requiring auth), but what do we gain from that vs the current approach?

I understand the desire to reduce the repetition of the scripting but I also see value in having the ability to have each "bundle" specify how it should be built rather than trying to optimize for a single common script

Reply:
As discussed via Slack I created https://github.com/operator-framework/rukpak/issues/885 to track this as a follow up to this PR

Reply:
> but what do we gain from that vs the current approach?
Simplicity

In rukpak, a bundle is an `fs.FS`. So given that we're going for consistency and simplicity, it seems reasonable for us to have a Dockerfile like this that we can re-use for any bundle.

```Dockerfile
FROM scratch
ARG destBundleRoot=/
ADD bundle.tar.gz $destBundleRoot
````

And then the configmap would contain this Dockerfile entry and a `bundle.tar.gz` entry, which we could mount into a pod to do a generic build.